### PR TITLE
Feature/home ci

### PR DESCRIPTION
**Pull request by Dranaxel**
I created hidden scripts reproducing the circleci workflow, then we call them all with circleci script.

### CODE REVIEW

**Comment by bovine3dom** [comment]
This is great, thanks!

I think I'd like the scripts to be more visible. I would suggest putting them in a new folder in the root of the project, e.g. `ci/`, so that it's obvious to people when a CI job has failed which script to run to replicate it on their machine.

Also - `.circleci/config.yml` doesn't seem to have any changes to it, so CircleCI isn't actually using these scripts. My intention was that it should - otherwise they'll inevitably fall out of synchronisation when someone updates the job but forgets to update the script.

**Comment by Dranaxel** [comment]
Ok, I understand so you want circleci to call every script in each steps instead of a multi-line ?

**Comment by bovine3dom** [comment]
e.g. https://github.com/tridactyl/tridactyl/blob/f047e5c895eda2f4cc2c9feb071ee518cacbd3cf/.circleci/config.yml#L23 and the three lines below it would become `bash -c "ci/lint.sh"`

all the setup - installing files etc. - should remain in `.circleci/config.yml`.

File: ci/commoncmd.sh [hunk 1]
```diff
@@ -1,3 +1,2 @@
 #! /bin/sh
-yarn install --frozen-lockfile
 yarn run build
```
Comment (bovine3dom) [inline ci/commoncmd.sh:2]: I don't think you need to build stuff every time - IIRC only the mozilla job needs it

**Comment by bovine3dom** [comment]
I think you just need to do `chmod +x` on the scripts :)

File: ci/lint.sh [hunk 2]
```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+SCRIPT_PATH=${0%/*}
+if [ "$SCRIPT_PATH" = "$0" ]; then
+	cd ..
+fi
+./ci/commoncmd.sh
+yarn install --frozen-lockfile
```
Comment (bovine3dom) [inline ci/lint.sh:7]:
```suggestion
./ci/commoncmd.sh
yarn install --frozen-lockfile
```

I don't think you need either of these lines (certainly not the `yarn install` one as that's done in config.yml)

File: ci/lint.sh [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+if ! [ -x "$(command -v shellcheck)" ]; then
+	GLOBIGNORE="node_modules" shellcheck -e2012 **/*.sh
+else
+	echo "You don't have shellcheck installed, we cannot validate shell scripts"
```
Comment (bovine3dom) [inline ci/lint.sh:11]:
```suggestion
	echo "Warning: shellcheck is not installed, skipping shell scripts"
```

File: ci/lint.sh [hunk 4]
```diff
@@ -0,0 +1,14 @@
+#!/bin/sh
+SCRIPT_PATH=${0%/*}
+if [ "$SCRIPT_PATH" = "$0" ]; then
+	cd ..
+fi
```
Comment (bovine3dom) [inline ci/lint.sh:5]: This snippet is quite strange - it looks like you can only call it from `ci/` or the root folder.

Why not 

```suggestion
cd ${0%/*}/..
```
? IIRC that should always put you in the root folder.

File: .circleci/circle.sh [hunk 5]
```diff
@@ -0,0 +1,5 @@
+#! /bin/sh
```
Comment (bovine3dom) [inline .circleci/circle.sh:1]: I don't think this script works or is needed

File: .circleci/config.yml [hunk 6]
```diff
@@ -11,7 +11,7 @@ commands:
           key: dependency-cache-{{ checksum "package.json" }}
           paths:
             - ./node_modules
-      - run: yarn run build
+      - run: bash -c 'ci/commoncmd.sh'
```
Comment (bovine3dom) [inline .circleci/config.yml:14]: As I said below I don't think you need to build stuff every time, only for the mozilla job

**Pull request merged**